### PR TITLE
SAK-46034 Lessons Add templates for page layouts

### PR DESCRIPTION
**Pull request by jkozar2**
This PR contains a system that gives users the option to add multi-item page-layout templates to Lessons. There are currently three templates, each with a description and preview image; the system is easily expandable to include more templates in the future. The templates are modular; more than one can be put on a page, and a newly-added one will tack onto the bottom of a page without affecting what's already present.

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (ern) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:1157]: lombok annotations will make this boiler plate go away!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (bjones86) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4761]: Can you provide i18n translations for this?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (bjones86) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4766]: And here, and any others that may have been missed

**Review by bjones86 — CHANGES_REQUESTED** [review]

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:405]: Could we use an ENUM here?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4702]: Could we use a switch instead using the ENUM?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:406]: Not a big fan of this, could we maybe use something like IntStream.range(1, 21).toArray()

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4707]: Could we use constants for "failure"?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4712]: Boolean.FALSE.toString()

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4735]: Boolean.FALSE.toString()

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4712]: Let's define a constant FORCE_BTN and reuse it

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4733]: SECTION constant

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4734]: Use constants

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4742]: Is this hardcoded in English? I'm fine raising I18N or L10N bugs after merging this.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4755]: Use constants

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:4761]: Is this hardcoded in English? I'm fine raising I18N bugs later.

File: lessonbuilder/tool/src/webapp/js/show-page.js
Comment (mpellicer) [inline lessonbuilder/tool/src/webapp/js/show-page.js:3986]: Could we use switch instead?

**Comment by mpellicer** [comment]
I made around 13 comments but honestly, it's about source and not about behavior, I really appreciate this amazing contribution @jkozar2 and I'm not going to block the merge for my comments!

@ern I don't think we should block the PR for our comments, if they can address them better but definitelly a good contribution!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java
Comment (jkozar2) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java:406]: I have remade the array where it gets used, in ShowPageProducer.createPageLayoutTab, with a loop if that works better.

**Review by bjones86 — APPROVED** [review]

**Pull request merged**
